src/routes/auth/decorators/auth-payload.decorator.ts

```diff
@@ -0,0 +1,13 @@
+import { createParamDecorator, ExecutionContext } from '@nestjs/common';
+import { Request } from 'express';
+import { AuthPayload } from '@/domain/auth/entities/auth-payload.entity';
+
+/**
+ * Route decorator that extracts the {@link AuthPayload}
+ */
+export const AuthPayloadDecorator = createParamDecorator(
```

Comment:
Nit: is it maybe a bit redundant to use `Decorator` in the decorator name? What about simply `@Auth`? 

(On [the NestJS documentation](https://docs.nestjs.com/custom-decorators) the examples have simpler names also, e.g.: `export const User = createParamDecorator(...)`)

Reply:
I used `RouteUrlDecorator` as a reference but I agree in hindsight. I've renamed it in 3d66d93afa832080bb9869adca7f1da9442676d5.